proof_of_concept/rest/internal_client.py

```diff
@@ -52,3 +59,74 @@ def add_rule(self, rule: Rule) -> None:
         r = requests.post(f'{self._endpoint}/rules', json=serialize(rule))
         if r.status_code != 200:
             raise RuntimeError(f'Error adding rule to site: {r.text}')
+
+    def submit_job(self, job: Job) -> str:
+        """Submits a job to the DDM via the local site.
+
+        Args:
+            job: The job to execute.
+
+        Returns:
+            The new job's id.
+
+        """
+        r = requests.post(
+                f'{self._endpoint}/jobs', json=serialize(job),
+                params={'requester': self._site}, allow_redirects=False)
+        if r.status_code != 303:
+            raise RuntimeError(f'Error submitting job: {r.text}')
+        if 'location' not in r.headers:
+            raise RuntimeError('Invalid server response when submitting job')
+        job_uri = r.headers['location']
+        prefix = f'{self._endpoint}/jobs/'
+        if not job_uri.startswith(prefix):
+            raise RuntimeError(
+                    'Unexpected server response when submitting job')
+        return job_uri
+
+    def is_job_done(self, job_id: str) -> bool:
```

Comment:
I was thinking to hide the fact that it's a URI from the user. We could also extract that number here and pass that, which is actually what I had until I realised that it was easier to just pass the URI itself. But I'd like for this to be an opaque handle. Maybe I should call it a job handle (since we also have the Identifier class, which this is definitely not) but anything called handle always sounds so Windowsy... Hmm.

Reply:
Hmmm, ok. I think for now this is fine then, I was able to figure out that it was a uri so I guess any other contributor could (considering I am probably the laziest contributor you will ever encounter)